mozdownload/scraper.py

```diff
@@ -311,19 +311,15 @@ def _download():
                 if log_level <= logging.INFO and content_length:
                     pbar.finish()
 
-            except requests.exceptions.HTTPError as ex:
+            except Exception as ex:
                 if os.path.isfile(tmp_file):
                     os.remove(tmp_file)
-                if ex.response.status_code == 404:
-                    raise errors.NotFoundError("The requested file was not found", self.url)
+                if type(ex) == requests.exceptions.HTTPError:
+                    if ex.response.status_code == 404:
```

Comment:
Please use `is` to compare the exception types, and also combine both if conditions into a single one. Right now we won't raise any exception if it's of type `HTTPError` and if status is not 404.

Reply:
I have made this change